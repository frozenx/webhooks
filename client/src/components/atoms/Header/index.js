import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import Header from '@ep/header';


export class HeaderComponent extends Component {
    AltLink = ({ children, href }) => {
        return <Link to={href}>{children}</Link>;
    }


    render() {
        return (
            <React.Fragment>
                <Header
                    primaryNavigationList={[{
                        id: "createsupplier",
                        altLink: () => (
                            <Link to="/tradingpartner/manage" id="createsupplier">
                                Create new partner
                          </Link>
                        ),
                        href: "/tradingpartner/manage",
                        text: "Create new partner"
                    },
                    {
                        id: "managesupplier",
                        href: "/tradingpartner",
                        altLink: () => (
                            <Link to="/tradingpartner" id="managesupplier">
                                Manage partners
                          </Link>
                        ),
                        text: "Manage partners"
                    }]}
                    logoSubtitle={"Trading Partner Service"}
                    altLink={this.AltLink}
                    secondaryNavigationList={[]}
                    showNavigationMenus={true}
                />


            </React.Fragment >
        )
    }
}
export default HeaderComponent;





