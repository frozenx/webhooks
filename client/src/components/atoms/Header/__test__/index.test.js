import React from 'react';
import Header from '..';
import {shallow} from 'enzyme';

const getProps = (additionalProps = {}) => {
  return {
    userName: 'Tesco UserName',
    signOutHandler: jest.fn(),
    name: "OAuthAccessToken",
    ...additionalProps
  };
}

const setup = (additionalProps = {}) =>
  shallow(<Header 
      {...getProps(additionalProps)} 
  />
);

describe('`Header` component', () => {

  it('should render without crashing..', () => {
      const wrapper = setup();

      wrapper.debug();
      expect(wrapper).toMatchSnapshot();
  });

  it('should render the username', async () => { 
    global.fetch = jest.fn().mockImplementation(() => ({
      status: 200,
      json: () => new Promise((resolve, reject) => {
        resolve(({firstName: "John", lastName: "Doe"}))
      }),
    }))
    const wrapper = await shallow(<Header/>)
    await wrapper.update()
    expect(wrapper.find("_default").props().appbar.props.menu[0].text).toBeTruthy()
  })

  it('should throw an error', async () => { 
    global.fetch = jest.fn().mockImplementation(() => ({
      json: () => new Promise((resolve, reject) => {
        reject({message: "error occured"})
      }),
    }))
    const wrapper = await shallow(<Header/>)
    await wrapper.update()
    const textShallow = shallow(wrapper.find("_default").props().appbar.props.menu[0].text)
    expect((textShallow.find(".loading-dots").length)).toBe(1);
  })

  it('should throw internal error ', async () => { 
    global.fetch = jest.fn().mockImplementation(() => ({
      status: 500,
    }))
    const wrapper = await shallow(<Header/>)
    await wrapper.update()
    const textShallow = shallow(wrapper.find("_default").props().appbar.props.menu[0].text)
    expect((textShallow.find(".loading-dots").length)).toBe(1);
  })

  it('should expose signout',  () => { 
    const wrapper =  shallow(<Header/>)
    const shallowAppBar = shallow(wrapper.find("_default").props().appbar)
    window.location.assign = jest.fn();
    shallowAppBar.props().menu[1].onClick()
    expect(window.location.assign).toHaveBeenCalledWith('https://toolkit.tesco.com/sign-in/');
    window.location.assign.mockRestore();
  })

});


