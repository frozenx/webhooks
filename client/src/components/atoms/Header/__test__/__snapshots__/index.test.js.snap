// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`Header\` component should render without crashing.. 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HeaderComponent
    name="OAuthAccessToken"
    signOutHandler={[MockFunction]}
    userName="Tesco UserName"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <_default
        appbar={
          <AppBar
            menu={
              Array [
                Object {
                  "className": "app-name",
                  "id": "appbar-usename",
                  "text": <dotLoader />,
                },
                Object {
                  "className": "app-name",
                  "hasPopup": true,
                  "id": "appbar-signout",
                  "onClick": [Function],
                  "text": "Sign out",
                },
              ]
            }
          />
        }
        masthead={
          <_default
            branding={
              Object {
                "href": "/tesco.com",
                "logo": Object {
                  "height": Object {
                    "desktop": "35px",
                    "mobile": "26px",
                  },
                  "width": Object {
                    "desktop": "121px",
                    "mobile": "88px",
                  },
                },
                "signpost": "Supplier",
              }
            }
          />
        }
        primaryNavigation={<SecondaryNavigation />}
        root={true}
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "appbar": <AppBar
          menu={
            Array [
              Object {
                "className": "app-name",
                "id": "appbar-usename",
                "text": <dotLoader />,
              },
              Object {
                "className": "app-name",
                "hasPopup": true,
                "id": "appbar-signout",
                "onClick": [Function],
                "text": "Sign out",
              },
            ]
          }
        />,
        "masthead": <_default
          branding={
            Object {
              "href": "/tesco.com",
              "logo": Object {
                "height": Object {
                  "desktop": "35px",
                  "mobile": "26px",
                },
                "width": Object {
                  "desktop": "121px",
                  "mobile": "88px",
                },
              },
              "signpost": "Supplier",
            }
          }
        />,
        "primaryNavigation": <SecondaryNavigation />,
        "root": true,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <_default
          appbar={
            <AppBar
              menu={
                Array [
                  Object {
                    "className": "app-name",
                    "id": "appbar-usename",
                    "text": <dotLoader />,
                  },
                  Object {
                    "className": "app-name",
                    "hasPopup": true,
                    "id": "appbar-signout",
                    "onClick": [Function],
                    "text": "Sign out",
                  },
                ]
              }
            />
          }
          masthead={
            <_default
              branding={
                Object {
                  "href": "/tesco.com",
                  "logo": Object {
                    "height": Object {
                      "desktop": "35px",
                      "mobile": "26px",
                    },
                    "width": Object {
                      "desktop": "121px",
                      "mobile": "88px",
                    },
                  },
                  "signpost": "Supplier",
                }
              }
            />
          }
          primaryNavigation={<SecondaryNavigation />}
          root={true}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "appbar": <AppBar
            menu={
              Array [
                Object {
                  "className": "app-name",
                  "id": "appbar-usename",
                  "text": <dotLoader />,
                },
                Object {
                  "className": "app-name",
                  "hasPopup": true,
                  "id": "appbar-signout",
                  "onClick": [Function],
                  "text": "Sign out",
                },
              ]
            }
          />,
          "masthead": <_default
            branding={
              Object {
                "href": "/tesco.com",
                "logo": Object {
                  "height": Object {
                    "desktop": "35px",
                    "mobile": "26px",
                  },
                  "width": Object {
                    "desktop": "121px",
                    "mobile": "88px",
                  },
                },
                "signpost": "Supplier",
              }
            }
          />,
          "primaryNavigation": <SecondaryNavigation />,
          "root": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
